addons/web_tour/static/src/js/tour_service.js

```diff
@@ -33,25 +34,98 @@ return session.is_bound.then(function () {
+                    record: record,
+                    oldValue: record.oldValue || '',
+                    newValue: newValue || '',
+                });
+            }
+            mutationTimer = setTimeout(() => _processMutations(), 500);
+        });
+        const tooltipParentRegex = /\bo_tooltip_parent\b/;
+        function _processMutations() {
+            nextMutation:for (const mutation of currentMutations) {
+                // First check if the mutation applied on an element we do not
+                // track (like the tour tips themself).
+                if (!_isTracked(mutation.record.target)) {
+                    continue nextMutation;
+                }
+
+                if (mutation.record.type === 'childList') {
+                    // If it is a modification to the DOM hierarchy, only
+                    // consider the addition/removal of tracked nodes.
+                    for (const nodes of [mutation.record.addedNodes, mutation.record.removedNodes]) {
+                        for (const node of nodes) {
+                            if (!_isTracked(node)) {
+                                continue nextMutation;
+                            }
+                        }
+                    }
+                } else if (mutation.record.type === 'attributes') {
+                    const oldV = mutation.oldValue.trim();
+                    const newV = mutation.newValue.trim();
 
-                function is_untracked(node) {
-                    var record_class = node.className;
-                    return (_.isString(record_class) &&
-                        _.intersection(record_class.split(' '), untracked_classnames).length !== 0);
+                    // Not sure why but this occurs, especially on ID change
+                    // (probably some strange jQuery behavior, see below).
+                    // Also sometimes, a class is just considered changed while
+                    // it just loses the spaces around the class names.
+                    if (oldV === newV) {
+                        continue nextMutation;
+                    }
+
+                    if (mutation.record.attributeName === 'id') {
+                        // Check if this is not an ID change done by jQuery for
+                        // performance reasons.
+                        if (oldV.includes('sizzle') || newV.includes('sizzle')) {
+                            continue nextMutation;
+                        }
+                    } else if (mutation.record.attributeName === 'class') {
+                        // Check if the change is about receiving or losing the
+                        // 'o_tooltip_parent' class, which is linked to the tour
+                        // service system.
+                        const hadClass = tooltipParentRegex.test(oldV);
+                        const hasClass = tooltipParentRegex.test(newV);
+                        if (hadClass !== hasClass) {
```

Comment:
you may loose another change here, but I guess it's only theoretical and it won't happen

Reply:
As discussed, see next point :+1: